lib/puppet/provider/rh_subscription/redhat.rb

```diff
@@ -1,41 +1,40 @@
 # @author Craig Dunn crayfishx/puppet-rhsm
+# @author Stefan Peer speer/puppet-rhsm
+#   (do not run subscription-manager on every Puppet run)
 Puppet::Type.type(:rh_subscription).provide(:redhat) do
   commands rhsm: '/usr/sbin/subscription-manager'
+  commands rct: '/usr/bin/rct'
+
   mk_resource_methods
 
   def self.subscriptions
     subs = {}
-    rhsm('list', '--consumed').split("\n\n").each do |blk|
+    Dir.glob('/etc/pki/entitlement/*.pem') do |cert|
+      next if cert =~ %r{-key.pem$}
+      cert_raw_data = rct('cat-cert', cert)
       sub_data = {}
-      blk.split("\n").each do |line|
-        element = line.match(%r{^([^:]+):[^\w]+(\w.*)})
-        next unless element.is_a?(MatchData)
-        sub_data[element[1]] = element[2]
+      section = nil
+      cert_raw_data.split("\n").each do |line|
+        if line =~ %r{^([^:\s]+):$}
```

Comment:
what are you doing here?

Reply:
maybe some example of the output of rct would help.

Reply:
Thank's for your feedback! Added some documentation, hope that explains it better :)